monitor/utils/events.js

```diff
@@ -96,11 +97,40 @@ async function main(mode) {
       }
     })).map(normalizeEvent)
 
+    const foreignBridgeSwapped = new web3Foreign.eth.Contract([TOKENS_SWAPPED_EVENT_ABI], COMMON_FOREIGN_BRIDGE_ADDRESS)
+    const tokensSwappedEvents = (await getPastEvents(foreignBridgeSwapped, {
+      event: 'TokensSwapped',
+      fromBlock: MONITOR_FOREIGN_START_BLOCK,
+      toBlock: foreignBlockNumber
+    })).map(normalizeEvent)
+
+    // Get transfer events for each previous erc20
+    await Promise.all(
+      tokensSwappedEvents.map(async swap => {
+        const previousERC20 = new web3Foreign.eth.Contract(ERC20_ABI, swap.recipient)
+
+        const previousTransferEvents = (await getPastEvents(previousERC20, {
+          event: 'Transfer',
+          fromBlock: MONITOR_FOREIGN_START_BLOCK,
+          toBlock: foreignBlockNumber,
+          options: {
+            filter: { to: COMMON_FOREIGN_BRIDGE_ADDRESS }
+          }
+        })).map(normalizeEvent)
+        transferEvents = [...previousTransferEvents, ...transferEvents]
+      })
+    )
+
     // Get transfer events that didn't have a UserRequestForAffirmation event in the same transaction
-    const directTransfers = transferEvents.filter(
+    let directTransfers = transferEvents.filter(
       e => foreignToHomeRequests.findIndex(t => t.referenceTx === e.referenceTx) === -1
     )
 
+    // filter transfer that is part of a token swap
+    directTransfers = directTransfers.filter(
+      e => tokensSwappedEvents.findIndex(t => t.referenceTx === e.referenceTx) === -1
```

Comment:
In this case `TokensSwapped` will affect an occurrence of `Transfer` even if `Transfer` is not a result of `mint` call. Consider to check the zero address in the field `from` of the corresponding `Transfer` at least.

Reply:
Thanks, added the check for `from` parameter